### PR TITLE
Add support for --use option

### PR DESCRIPTION
**Pull request by RyanZim**
Attn: @ai @michael-ciniawsky

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -146,6 +129,27 @@ function processFile (file, config, watcher) {
   })
 }
 
+function config () {
```
Comment (michael-ciniawsky) [inline index.js]: 👍  If you check for `postcss.config.js` first, `-p|-m|-s|-t` (options in conditional) could be used as context if set and config is present, but can be done later/should be non breaking

File: index.js [hunk 2]
```diff
@@ -146,6 +129,27 @@ function processFile (file, config, watcher) {
...
+  if (argv.use || argv.parser || argv.stringifier || argv.syntax) {
+    return {
+      plugins: argv.use ? argv.use.map(i => require(i)) : [],
```
Comment (michael-ciniawsky) [inline index.js]: `i` => `plugin`

File: index.js [hunk 3]
```diff
@@ -146,6 +129,27 @@ function processFile (file, config, watcher) {
...
+      options: {
+        parser: argv.parser ? require(argv.parser) : undefined,
+        syntax: argv.syntax ? require(argv.syntax) : undefined,
+        stringifier: argv.stringifier ? require(argv.stringifier) : undefined,
+        map: argv.map
+      }
+    }
+  } else {
+    return postcssrc()
+    .catch(e => {
```
Comment (michael-ciniawsky) [inline index.js]: `e` =>  `err`

```js
return postcssrc()
  .catch(err => err.message.indexOf('...') === -1 
     ? throw err 
     : { plugins: [], options: {} }
  )
```

Comment (RyanZim) [inline index.js]:
```
standard: Use JavaScript Standard Style (http://standardjs.com)
  /home/ryan/postcss-cli/index.js:146:10: Parsing error: Unexpected token throw
```
Will rename `e` => `err`, though.

Comment (michael-ciniawsky) [inline index.js]: Then my linter f** up ) I was wondering, the `{ return err}` are needed , anyways just cosmetics :)

**Comment by ai** [comment]
Did we forget `--use` docs?

**Comment by michael-ciniawsky** [comment]
@ai Yep good catch, but I like to refactor them a bit anyways and will add it then

**Comment by RyanZim** [comment]
Docs need expanded all around; @michael-ciniawsky are you on that?

**Comment by michael-ciniawsky** [comment]
@RyanZim yes, when feature completed I will upate docs

**Pull request merged**